bots/checkout/src/main/java/org/openjdk/skara/bots/checkout/CheckoutBot.java

```diff
@@ -135,6 +135,7 @@ public String toString() {
                     var toRepo = Repository.get(to).orElseThrow(() ->
                         new IllegalStateException("Repository vanished from " + to));
                     var existing = new ArrayList<>(marks.current());
+                    Collections.sort(existing);
```

Comment:
If the `convert` method relies on the marks being sorted, maybe it would be more robust to move the call to sort into there?

Reply:
Yeah, I agree. Problem is that the `convert` method takes a `List<Mark>` (which might be an `UnmodifiableList`), so I can't just pass it to `Collections.sort` :disappointed: I would have to make a copy of the passed list and then sort _that_ list, and that just seems a bit annoying (give that we can have around 100,000 marks).